### ISSUE TITLE
PARENT ISSUE: Fix the usage of userType

### ISSUE THREAD

**Issue by palisadoes** [PalisadoesFoundation/talawa-api#1965]
**Describe the bug**

1. The schema doesn't apply a person's `userType` by organization. 
1. The `userType` is currently a part of the `User` definition not part of a `userOrganization` definition or something similar. 
    1. This means that an `Admin` is an `Admin` for all organizations which grants the same rights as a `SuperAdmin`. This is not the intention.
    2. This has created historical confusion in the `Admin` and `SuperAdmin` roles.

**Related Issues**

1. https://github.com/PalisadoesFoundation/talawa-api/issues/1711
1. https://github.com/PalisadoesFoundation/talawa/issues/2373
1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1445
1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1451
1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1452
1. https://github.com/PalisadoesFoundation/talawa-api/issues/1862
1. https://github.com/PalisadoesFoundation/talawa-api/issues/1904
1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1445
1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1682
1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1683
1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1684
1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1685
1. https://github.com/PalisadoesFoundation/talawa-admin/issues/1686
1. https://github.com/PalisadoesFoundation/talawa-api/issues/1987

**Comment by palisadoes** [comment]
@kanhaiya04
@Atharva-Kanherkar
@adi790uu
@Shubh152

1. When do you think we will be ready to merge  the `develop-userTypeFix` branch back into `develop`?
2. There are only a few issues left
3. Would the weekend be possible with a short freeze on merging PRs? Any other date in the near future?

**Comment by Atharva-Kanherkar** [comment]
@palisadoes I can try over the weekend, I have no other commitments. I am still trying to research a previous issue, so the weekend seems like a good deadline. Will try best possible. approach as soon as I can with my capabilities.

**Comment by kanhaiya04** [comment]
@palisadoes I will try to complete as soon as possible.

**Comment by AVtheking** [comment]
@Atharva-Kanherkar you can ask me if you have any doubt in the api as I have worked on that .

**Comment by palisadoes** [comment]
Do you think we'll be able to merge back into develop on Sunday night IST?

**Comment by Atharva-Kanherkar** [comment]
> Do you think we'll be able to merge back into develop on Sunday night IST?

There are a lot of merge conflicts and i am working on my best to resolve them and maintain the current functionality. While i can not promise the exact time, i can try my best to do it sunday night IST or monday morning.

**Comment by Shubh152** [comment]
@palisadoes I have pushed my changes on PR #2013

**Comment by palisadoes** [comment]
> > Do you think we'll be able to merge back into develop on Sunday night IST?
> 
> There are a lot of merge conflicts and i am working on my best to resolve them and maintain the current functionality. While i can not promise the exact time, i can try my best to do it sunday night IST or monday morning.

Why are there merge conflicts when you merged the PR into this branch without them?

**Comment by palisadoes** [comment]
> @palisadoes I have pushed my changes on PR #2013

How will this impact the merge conflict problem previously mentioned?

**Comment by Atharva-Kanherkar** [comment]
> > > Do you think we'll be able to merge back into develop on Sunday night IST?
> > 
> > 
> > There are a lot of merge conflicts and i am working on my best to resolve them and maintain the current functionality. While i can not promise the exact time, i can try my best to do it sunday night IST or monday morning.
> 
> Why are there merge conflicts when we merged the PR into this branch without them?

I do not know, but i have resolved a lot of them, but there are still a lot of conflicts.. Was this not supposed to happen? I thought it was normal to have these conflicts, so i started fixing them? But there seems to be a lot of them in number.

**Comment by palisadoes** [comment]
I wasn't sure how far back in the merged PR chain you were updating. It sounds like you are doing up to the most current one. This helps.

I'll try not to merge any more PRs into any of the repos so that we can have stability until we are ready to merge back into `develop`. 

I'll let the other PR submitters know to expect a delay over the weekend in a few hours

**Comment by Shubh152** [comment]
> > @palisadoes I have pushed my changes on PR #2013
> 
> How will this impact the merge conflict problem previously mentioned?

I think it should not introduce more merge conflicts.

**Comment by Atharva-Kanherkar** [comment]
It is still causing merge conflicts. Is the PR merged? So that i can be sure that it is not causing any merge conflicts?

**Comment by Shubh152** [comment]
> It is still causing merge conflicts. Is the PR merged? So that i can be sure that it is not causing any merge conflicts?

No, it is not merged yet.

**Comment by AVtheking** [comment]
> > @palisadoes I have pushed my changes on PR #2013
> 
> How will this impact the merge conflict problem previously mentioned?

@palisadoes  it's because he has merged the latest changes in the develop branch which will cause merge conflicts in on all the files that are being modified in the develop recently and in develop-UserTypeFix branch

**Comment by palisadoes** [comment]
> > It is still causing merge conflicts. Is the PR merged? So that i can be sure that it is not causing any merge conflicts?
> 
> No, it is not merged yet.

The tests are failing. Please fix

**Comment by palisadoes** [comment]
@AVtheking @Atharva-Kanherkar can you chat on slack to figure out what is going on with the conflicts.

No new PRs have been merged for 2 days.

**Comment by AVtheking** [comment]
@palisadoes  yes we are interacting on slack to resolve the conflicts

**Comment by palisadoes** [comment]
> @palisadoes  yes we are interacting on slack to resolve the conflicts

@Atharva-Kanherkar @AVtheking 

We've stopped merging PRs for five days to facilitate this. The queue is getting long.

Do you have an estimate as to when this will be merged?

**Comment by AVtheking** [comment]
@palisadoes  @Atharva-Kanherkar is resolving test failures and some errors ,I am trying my best to help him ,he would soon raise a PR.

**Comment by Atharva-Kanherkar** [comment]
@AVtheking I will raise a PR with the failing tests, can you please review them and tell me whats wrong? I guess it will be easier for you and me both to faciliate this way.

**Comment by palisadoes** [comment]
Yes, that would be helpful. Everyone would be able to assist.

**Comment by Atharva-Kanherkar** [comment]
> Yes, that would be helpful. Everyone would be able to assist.

I will make the PR shortly!

**Comment by Atharva-Kanherkar** [comment]
@palisadoes 
<img width="942" alt="image" src="https://github.com/PalisadoesFoundation/talawa-api/assets/142440039/a671879a-1c0a-4805-bae3-9e074585d80c">
Are these commits coming because i merged all the changes from the develop branch, is it fine?
<img width="936" alt="image" src="https://github.com/PalisadoesFoundation/talawa-api/assets/142440039/050e48aa-28a1-4875-9be7-ffd424c0cae7">

**Comment by Shubh152** [comment]
@AVtheking @Atharva-Kanherkar Let me know if you need any assistance from me.

**Comment by palisadoes** [comment]
Yes, those commits are OK.

**Comment by palisadoes** [comment]
@AVtheking
@kanhaiya04
@Atharva-Kanherkar
@adi790uu
@Shubh152

1. We have a real problem here with the last PR
    1. https://github.com/PalisadoesFoundation/talawa-api/pull/2032
1. We merged past PRs assuming tests would be completed later. The issue now is that we need all the tests to pass before we can merge into `develop`. There are are 500 references to userType in the `tests` directory that need to be rectified.

We can try to remove the references in `src/` but the `tests/` directory needs work.

How can we split up the work to fix these tests?

**Comment by AVtheking** [comment]
@palisadoes I don't think this a lot of work as because it also searched TestUserType along with userType which is just to know whether the user is going to be admin/superAdmin or not

**Comment by AVtheking** [comment]
@palisadoes what he does is accepted the new features from the develop branch and not refactored them , and those features have refrence to userType field

**Comment by palisadoes** [comment]
1. The spreadsheet is ready. Only one person has signed up.
2. Let's see how we can complete this and merge it back into develop in the next 2-3 days.
